internal/engine/dockerprune/docker_pruner.go

```diff
@@ -35,57 +36,77 @@ func NewDockerPruner(dCli docker.Client) *DockerPruner {
 	return &DockerPruner{dCli: dCli}
 }
 
-func (dp *DockerPruner) DisableForTesting() {
-	dp.disabledForTesting = true
+func (dp *DockerPruner) DisabledForTesting(disabled bool) {
+	dp.disabledForTesting = disabled
 }
 
 func (dp *DockerPruner) OnChange(ctx context.Context, st store.RStore) {
 	if dp.disabledForTesting {
 		return
 	}
 
-	if dp.started {
-		return
-	}
-
 	state := st.RLockState()
 	defer st.RUnlockState()
 
-	if state.DockerPruneSettings.Disabled {
+	settings := state.DockerPruneSettings
+	if settings.Disabled {
 		return
 	}
 
 	if err := dp.sufficientVersionError(); err != nil {
 		return
 	}
 
-	// wait until state has been kinda initialized, and there's at least one Docker build
-	if !state.TiltStartTime.IsZero() && state.LastTiltfileError() == nil && state.HasDockerBuild() {
-		dp.started = true
-		go func() {
-			select {
-			case <-time.After(time.Minute * 2):
-				dp.Prune(ctx, defaultMaxAge) // report once pretty soon after startup...
-			case <-ctx.Done():
-				return
-			}
-
-			for {
-				select {
-				case <-time.After(dockerPruneInterval):
-					// and once every <interval> thereafter
-					dp.Prune(ctx, defaultMaxAge)
-				case <-ctx.Done():
-					return
-				}
-			}
-		}()
+	// If user doesn't have at least one Docker build, they probably don't care about pruning
+	if !state.HasDockerBuild() {
+		return
+	}
+
+	// Don't prune WHILE we're building something, in case of weird side-effects.
+	// TODO: more robust check here
+	if state.CurrentlyBuilding != "" {
```

Comment:
i'm pretty sure this won't behave as you expect. this will just fire as soon as the first build completes. i would probably use BuildController's needsBuild function